upup/pkg/fi/cloudup/template_functions.go

```diff
@@ -127,7 +127,7 @@ func (tf *TemplateFunctions) AddTo(dest template.FuncMap, secretStore fi.SecretS
 	dest["GossipDomains"] = func() []string {
 		var names []string
 
-		if cluster.IsGossip() {
+		if dns.IsGossipClusterName(cluster.Name) {
```

Comment:
I don't understand this. If running NoneDNS in a cluster with a gossip name, `IsGossip()` is going to be false, yet all of the addons are going to enable the Gossip machinery?

Reply:
This is not quite needed in general, only for the `[sig-auth] ServiceAccounts ServiceAccountIssuerDiscovery should support OIDC discovery of service account issuer [Conformance]` test to pass.